GitHub for text independent

sycyhy opened the pull request.

Commit:
Fix code format

Commit:
Git Ignore

Commit:
VPTI-66

Refactor of flow client

Commit:
VPTI-66

FlowClient integration test

Commit:
VPTI-66

ExceptionMapper, refactor

Commit:
* added axion-release plugin
* added revision to manifest

Commit:
VPTI-89

Changed media types of requests multipartFormData to application/octet-stream

Commit:
Merge remote-tracking branch 'origin/feature/VPTI-66-flow-client' into feature/VPTI-66-flow-client

Commit:
Merge pull request #1 in VPTI/flow-client from feature/VPTI-66-flow-client to master

* commit '6f52e29a16ac6bcf5d1f707f0636abe856b9e26b':
  VPTI-89
  VPTI-89
  * added axion-release plugin * added revision to manifest
  VPTI-66
  VPTI-66
  VPTI-66
  Fix code format
  VPTI-66

Commit:
VPTI-89

Code formatting

Commit:
* added maven gradle plugin

Commit:
* configured project name

Commit:
* Add temporarly methods for returning results during verification process

Commit:
VFD-37 - add verification to flow-client

* Add verification process which allows getting current result.

Commit:
* Fix log4j depedency

Commit:
Fix typo

Commit:
VPTI-105

Add https authentication

Commit:
Merge pull request #2 in VPTI/flow-client from feature/VFD-5-verifying-speakers to master

* commit 'ffa16369f545e4ce191a13cd2e787b61a6fc991e':
  Fix typo
  * Fix log4j depedency
  VFD-37 - add verification to flow-client
  * Add temporarly methods for returning results during verification process

Commit:
VPTI-104
Reformat

Commit:
Merge branch 'master' into feature/VPTI-104-add-https-authentication-to-flow-client

Conflicts:
	src/main/java/com/voicepin/flow/client/Caller.java
	src/main/java/com/voicepin/flow/client/FlowClient.java

Commit:
VPTI-104
Rearrange import order after Luke

Commit:
* Change classes and methods access level modifiers
* Make classes or fields final if possible

Commit:
* Move log4j implementantion to the testCompile

Commit:
VPTI-104
Refactor. Removed checked exceptions from constructor.

Commit:
Merge remote-tracking branch 'origin/feature/VPTI-104-add-https-authentication-to-flow-client' into feature/VPTI-104-add-https-authentication-to-flow-client

Commit:
VPTI-110
Added getVoiceprint method.

Commit:
VPTI-112 - Implement

* Create Builder for FlowClient
* Allow to use https in three modes, unsafe, pinned and default

Commit:
VPTI-110
Javadoc update

Commit:
Merge pull request #3 in VPTI/flow-client from feature/VPTI-104-add-https-authentication-to-flow-client to master

* commit '8c8b7dda9566c049fc0f2cb44799cc0d0f732725':
  VPTI-104 Refactor. Removed checked exceptions from constructor.
  * Move log4j implementantion to the testCompile
  * Change classes and methods access level modifiers * Make classes or fields final if possible
  VPTI-104 Rearrange import order after Luke
  VPTI-104 Reformat
  VPTI-105

Commit:
* Fix typo in test

Commit:
VPTI-110
Refactor

Commit:
Merge pull request #5 in VPTI/flow-client from feature/VPTI-109-security-levels to master

* commit 'c61c0023f89adbe635764a5a3ba687819b7a2773':
  VPTI-110 Refactor
  * Fix typo in test
  VPTI-112 - Implement

Commit:
Merge branch 'master' into feature/VPTI-110-get-voiceprint-method-in-flow-client

Conflicts:
	src/main/java/com/voicepin/flow/client/VerifyStreamClient.java

Commit:
Merge pull request #4 in VPTI/flow-client from feature/VPTI-110-get-voiceprint-method-in-flow-client to master

* commit 'ac34316748a4bd91304f98bea8baced899ca1005':
  VPTI-110 Javadoc update
  VPTI-110 Added getVoiceprint method.

Commit:
* Add README.md file for github

kodrzywolek commented on src/integration-test/groovy/com/voicepin/flow/client/FlowClientIT.groovy:
```diff
@@ -0,0 +1,92 @@
+
+//        def url = "https://localhost:8443/voicepin-ti-server/v1/"
+//        client = FlowClient.newBuilder(url)
+//                .withHttps("test_deployment", "qwer")
```
To trochę słabe

Commit:
* Delete commented code in test

ludwix commented on README.md:
```diff
@@ -0,0 +1 @@
+# voicepin-flow-client
```
mozna by dodac jakies readme

ludwix commented on build.gradle:
```diff
@@ -0,0 +1,63 @@
+
+import org.ajoberstar.grgit.Grgit
+
+//available from axion-release
```
komentarz lepiej jakby był nad importem

Commit:
* Code review fixes

- Delete unnecessary json mappers
- Make fields final if possible
- Correct some method visibility

mckulpa commented on build.gradle:
```diff
@@ -0,0 +1,61 @@
+    id 'maven'
+    id 'pl.allegro.tech.build.axion-release' version '1.3.4'
+}
+//available from axion-release
```
usuńmy ten komentarz może

mckulpa commented on src/main/java/com/voicepin/flow/client/Caller.java:
```diff
@@ -62,27 +86,51 @@ protected Caller(final String baseURL) {
-    public void setInvocationBuilderFactory(final InvocationBuilderFactory invocationBuilderFactory) {
-        this.invocationBuilderFactory = invocationBuilderFactory;
-    }
+    <T> CompletableFuture<T> asyncCall(final Call<T> call) {
```
wyciągnąć część wspólną z call i asyncCall (nie zapomnieć że ClientProperties.REQUEST_ENTITY_PROCESSING powinno być w obydwu)

dodaj się do autorów

mckulpa commented on src/main/java/com/voicepin/flow/client/ExceptionMapper.java:
```diff
@@ -1,59 +1,59 @@
-                }
+class ExceptionMapper {
+
+    private final Collection<Definition> definitions = new ArrayList<>();
```
Stary sposób był szybszy, teraz musi iterować po każdym Definition i robić na każdym apply co też nie jest logiczne.
Rozumiem, że pozbyliśmy się mapy żeby np. wspierać też zakres, ale czy w takim wypadku nie wystarczyłoby dodać tego samego Definition z wieloma kluczami?

sycyhy replied:
" We follow two rules in the matter of optimization: Rule 1. Don’t do it. Rule 2 (for experts only). Don’t do it yet—that is, not until you have a perfectly clear and unoptimized solution. " :D

mckulpa replied:
No tak fajnie, ale nie jest napisane żeby obniżać wydajność :)
Rozumiem zwiększyć czytelność kosztem wydajności ale tutaj widzę straty po obydwu stronach

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+        private String username;
+        private String password;
+
+        private CertificateStrategy connectionHelper = new TrustedCertificateStrategy();
```
zakutalizować nazwę pola

mckulpa commented on src/main/java/com/voicepin/flow/client/VerifyStreamClient.java:
```diff
@@ -0,0 +1,67 @@
+ *
+ * @author Lukasz Warzecha
+ */
+public class VerifyStreamClient {
```
Może VerificationStream?

mckulpa commented on src/main/java/com/voicepin/flow/client/VerifyStreamClient.java:
```diff
@@ -0,0 +1,67 @@
+     * 
+     * @return current verification result
+     */
+    public VerifyResult getCurrentResult() throws FlowClientException {
```
A czemu blocking? nie może być async, skoro mamy to w Callerze?

sycyhy replied:
Moze i fajnie by bylo, ale nigdy taska nie mielismy a ciagnie to za soba zmiany czy to w demo czy w testach od Krzyska

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
 
 /**
- * @author kodrzywolek
+ * @author kodrzywolek, Lukasz Warzecha
```
Tu też by pasował jakiś krótki JavaDoc. Np:

Client for VoicePIN Flow REST API (Text Independent Biometric Voice Verification server).

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+     * Creates builder for FlowClient.
+     * 
+     * @param baseUrl url to the service eg.
+     *            http://flow.voicepin.com/voicepin-ti-server/v1/
```
zmienić adres na fikcyjny, np. localhost:8080
dodałbym też HTTPS

Commit:
* Code review fixes part2

- Rename some classes
- fix some javadocs

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
     }
 
+    /**
+     * Creates new Voiceprint, which is represented by UUID.
```
Proponuję (choć nie rozpisywałbym się tu za bardzo, dokładniejszy opis jest na swaggerze):

Creates a new Voiceprint. Returned ID should be passed to any subsequent operations on this Voiceprint (i.e. enrollment/verification).

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+     * 
+     * @param addVoiceprintRequest
+     * @return assigned voiceprint id
+     * @throws FlowClientException
```
Wyjątki trzeba udokumentować tak jak jest to w kliencie do FP (to się tyczy wszystkich metod, nie tylko tej). Tutaj FlowClientExcpetion jest tylko ich rootem, a w zależności od metod mogą polecieć np. InvalidAudio etc... Podobnie trzeba rozdzielić błąd połączenia od błędów logiki.

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+     * Creates new Voiceprint, which is represented by UUID.
+     * 
+     * @param addVoiceprintRequest
+     * @return assigned voiceprint id
```
Voiceprint ID (poprawić w całym projekcie)

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+     * Creates new voice model using streamed recording. Sets it for this Voiceprint. The model will be used as a
+     * reference in any subsequent verifications. If the Voiceprint has already been enrolled, its preexisting model
+     * would be overwritten.
+     * 
```
"Creates a new voice model using the provided stream and sets it for the Voiceprint with specified ID."

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+     * reference in any subsequent verifications. If the Voiceprint has already been enrolled, its preexisting model
+     * would be overwritten.
+     * 
+     * @param enrollRequest request with voiceprint id and recording stream
```
używajmy speech stream, poprawmy w całym projekcie

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
-    public VerifyResult verify(VerifyRequest verifyRequest) throws FlowClientException {
+    /**
+     * Starts verification process on given Voiceprint with incoming stream. Verification process starts immediately
+     * (i.e. it does not wait for the whole stream to be uploaded).
```
Nie wystarczy skopiować ze swaggera bo w kliencie role są trochę odwrócone i lepiej pasują inne słowa - incoming->provided, uploaded->sent
"Starts verification process on given Voiceprint with provided speech stream. Verification process starts immediately (i.e. it does not wait for the whole stream to be sent)."

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+     * (i.e. it does not wait for the whole stream to be uploaded).
+     *
+     * @param verifyRequest
+     * @return result client for getting results
```
ten komentarz i nazwa zwracanej klasy mi nie leżą, możemy obgadać.

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+         *
+         * @return
+         */
+        public FlowClientBuilder withHttps(String username, String password) {
```
Przecież HTTPS w standardzie nie wymaga autentykacji. To powinno być niezależne np. jako withBasicAuth(username, password).

sycyhy replied:
To odzwierciedla logike jaka jest zastosowana we flow-server, nie da sie tam uruchomic https bez basicAutha - a http w ogole nie umozliwia basicAutha

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,176 @@
+        }
+
+        /**
+         * Allows to connect to the Voicepin Flow server using secure http connection with
```
Voicepin->VoicePIN 
http->HTTPS

oprócze tego rozbić na 2 metody tak jak piszę niżej?

sycyhy replied:
to odzwierciedla logike jaka jest zastosowana we flow-server, nie da sie tam uruchomic https bez basicAutha - a http w ogole nie umozliwia basicAutha

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,179 @@
+        }
+
+        /**
+         * Allows Flow Client to trust all https certificates. Use with caution and only
```
HTTPS

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,179 @@
+        }
+
+        /**
+         * Allows to use custom self-signed certificate.
```
Nie tylko self-signed.

Allows certificate pinning - only certificates in passed key store will be accepted (self-signed included).

mckulpa commented on src/main/java/com/voicepin/flow/client/VerifyStream.java:
```diff
@@ -0,0 +1,68 @@
+ * Provides method for getting current result of verification process and setting listener
+ * on complete process.
+ *
+ * @author Lukasz Warzecha
```
Provides current verification results (while still streaming) and accepts a final success/error callback.

mckulpa commented on src/main/java/com/voicepin/flow/client/VerifyStream.java:
```diff
@@ -0,0 +1,68 @@
+
+    /**
+     * Adds listener for verification process. The listener will be called at the end of
+     * the verification process
```
Adds listener for verification process which will be called on final success or error.

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,179 @@
+    /**
+     * Starts verification process on given Voiceprint with incoming stream. Verification
+     * process starts immediately (i.e. it does not wait for the whole stream to be
+     * uploaded).
```
Starts verification process on given Voiceprint with provided speech stream. Verification starts immediately (i.e. it does not wait for the whole stream to be sent). The process is finished when the stream ends so in real-time verification it is crucial to listen to verification scores and end the stream when satisfying results are achieved instead of waiting for the final result.

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,179 @@
+     * uploaded).
+     *
+     * @param verifyRequest
+     * @return result client for getting results
```
masło maślane, już wolałbym nic nie pisać

mckulpa commented on src/main/java/com/voicepin/flow/client/FlowClient.java:
```diff
@@ -1,45 +1,179 @@
+     * @return result client for getting results
+     * @throws FlowClientException
+     */
+    public VerifyStream verify(VerifyRequest verifyRequest) throws FlowClientException {
```
może VerificationProcess jako nazwa klasy zwracanej?

mckulpa commented on src/main/java/com/voicepin/flow/client/VerifyListener.java:
```diff
@@ -0,0 +1,14 @@
+/**
+ * @author Lukasz Warzecha
+ */
+public interface VerifyListener {
```
może VerificationProcessListener?

Tak naprawdę może fajnie by było tutaj dać też wyniki częściowe zamiast osobnej metody.

sycyhy replied:
dalem TODO

mckulpa commented on src/main/java/com/voicepin/flow/client/VerifyStream.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    /**
+     * Perfoms blocking operation which asks Voicepin Flow server for the current
```
Returns current result while speech streaming may still be in progress.

mckulpa commented on src/main/java/com/voicepin/flow/client/data/SpeechStream.java:
```diff
@@ -1,16 +1,22 @@
  * @author mckulpa
  */
-public class SpeechStream extends InputStreamWrapper {
+public class SpeechStream extends InputStream {
```
MEGA WAŻNE

albo rezygnujemy z tej klasy albo wrapujemy wszystkie metody z InputStreama. Domyślnie w implementacji InputStream close nic nie robi, więc tego strumienia się nie zamknie, trzeba wywołać close na underlyingInputStream! Od razu warto dodać resztę metod, jest ich dosłownie kilka.

Commit:
- Code review fixes part3

* Add exceptions to java-docs
* Delete unused exceptions
* Rename classes
